### ISSUE TITLE
Resizing and 3.7 MB image is throwing an error

### ISSUE THREAD

**Issue by marian-ene** [LazZiya/ImageResize#3]
I've tried to resize a 3.7 MB image and I get "Out of memory.." error

**Comment by LazZiya** [comment]
Can you show some code and provide some more details?

- Which version you are using? 
- Do you this error with smaller images?
-  Any other details that helps to reproduce the issue?

**Comment by marian-ene** [comment]
```
using (var stream = item.OpenReadStream())
using (var img = Image.FromStream(stream))
using (var imgResized = ImageResize.ScaleAndCrop(img, 400, 300))
{
    imgResized.SaveAs(shortPaintingPath);
}
```
This is the code that I'm using. 
The version is 2.0.0.
I've attached the image to help with the debug,
On other smaller images work fine

![afis_PROMO67A2](https://user-images.githubusercontent.com/18363959/70545131-c648ba00-1b75-11ea-8f47-41bb1fee3a40.jpg)

**Comment by LazZiya** [comment]
The issue was due to image color format (CMYK), I've provided a fix for the issue in [e89d16f](https://github.com/LazZiya/ImageResize/commit/e89d16f341af06d5b4fb89a5d721892a81697c4b).

Now the image color format will be checked before resizing, if it is CMYK then the pixel color format will be changed to 32bppArgb, so a slightly color change will occor when resizing CMYK images.

The fix is included the latest preview version [v3.0.0-preview2](https://www.nuget.org/packages/LazZiya.ImageResize/3.0.0-preview2) available in nuget.

````
Install-Package LazZiya.ImageResize -Version 3.0.0-preview2
````
You can use the resizing code as below:
````cs
using (var img = Image.FromFile(@"wwwroot\images\my-image.jpg"))
{
    img.ScaleAndCrop(400, 300)
        .SaveAs(@"wwwroot\images\new-image.jpg");
}
````

Resized image: 
![new-image](https://user-images.githubusercontent.com/22244584/70861111-e0e6ae80-1f3a-11ea-8b15-d3f7c022cd6a.jpg)

Additionally adding image watermark and text watermark can be done as below:
````cs
using (var img = Image.FromFile(@"wwwroot\images\my-image.jpg"))
{
    var watermark = Image.FromFile(@"wwwroot\images\wm.png");

    img.ScaleAndCrop(400, 300)
        .AddImageWatermark(watermark)
        .AddTextWatermark("http://Ziyad.Info")
        .SaveAs(@"wwwroot\images\new-image.jpg");

    watermark.Dispose();
}
````

Final image:
![new-image](https://user-images.githubusercontent.com/22244584/70861190-8b5ed180-1f3b-11ea-8e33-69d5c64f2472.jpg)
